tests/unit/synapseclient/unit_test_commandline.py

```diff
@@ -593,16 +603,89 @@ def test_config_replace(
     mock__authenticate_login.return_value = "password"
     mock__replace_existing_config.return_value = "test"
 
-    expected_auth_section = "[authentication]\n" "username=test\n" "password=wow\n\n"
+    expected_auth_section = "[profile authentication]\nusername=test\npassword=wow\n\n"
     temp = tempfile.NamedTemporaryFile(mode="w", delete=False)
     args = Mock()
     args.configPath = temp.name
+    args.profile = "authentication"
     cmdline.config(args, syn)
     mock__replace_existing_config.assert_called_once_with(
-        temp.name, expected_auth_section
+        temp.name, expected_auth_section, "authentication"
     )
     temp.close()
 
+@patch("synapseclient.__main__.os.path.exists", return_value=False)
+@patch("synapseclient.__main__.open", new_callable=mock_open)
+@patch.object(cmdline, "_generate_new_config")
+@patch.object(cmdline, "_authenticate_login")
+@patch.object(cmdline, "_prompt_for_credentials")
+def test_config_generate_named_profile(
+        mock_prompt_for_credentials,
+        mock_authenticate_login,
+        mock_generate_new_config,
+        mock_exists,
+        syn,
+):
+    """Test config command generates a new config with a named profile"""
+    mock_prompt_for_credentials.return_value = ("testuser", "authtoken123")
+    mock_authenticate_login.return_value = "authtoken"
+    mock_generate_new_config.return_value = "config text"
+
+    args = Mock()
+    args.configPath = "test_config_path"
+    args.profile = "devprofile"
+
+    expected_auth_section = "[profile devprofile]\nusername=testuser\nauthtoken=authtoken123\n\n"
+    cmdline.config(args, syn)
+    mock_generate_new_config.assert_called_once_with(expected_auth_section, "devprofile")
```

Comment:
We should include assertions for everything that we expect to be called and also things we do not expect to be called. For example, we are currently missing an expected call of `_authenticate_login` and an assertion saying that we expect `_replace_existing_config` not to be called.